warehouse/packaging/views.py

```diff
@@ -84,11 +84,13 @@ def release_detail(release, request):
     if project.name != request.matchdict.get("name", project.name):
         return HTTPMovedPermanently(request.current_route_path(name=project.name))
 
+    # Prefer plain text if explicitly set, otherwise prefer html.
+    if release.description.content_type == "text/plain":
+        # Wrap as preformatted text to preserve whitespace.
+        description = f"<pre>{release.description.raw}</pre>"
```

Comment:
I think we probably need to do this at render-time instead and store it in `release.description.html` (either when we're rendering inline, like on the lines below, or when we're doing it in a task)

Reply:
Centralized the behavior in `readme.render()`.